src/jti_background.js

```diff
@@ -213,6 +254,80 @@ function addTemplate (templateName, issueTypeField, projectsField, text, callbac
     });
 }
 
+function addDomain (domainName, callback) {
+    chrome.storage.sync.get(StorageID, function (data) {
+        var domainJSON = {};
+
+        if (data[StorageID]) {
+            domainJSON = data[StorageID];
+        }
+
+        var domainID = getNextID(domainJSON.options.domains);
+        var newDomain = {
+            'id': domainID,
+            'name': domainName
+        };
+
+        validateDomain(domainName, function (message) {
+            if (message) {
+                callback(false, message);
+            } else {
+                domainJSON.options.domains[domainID] = newDomain;
+                // Refresh existing pages with this URL.
+                matchRegexToJsRegex(domainName);
+                chrome.tabs.query({windowId: chrome.windows.WINDOW_ID_CURRENT}, function (tabs) {
+                    $.each(tabs, function (tabIndex, tab) {
+                        var chromeRegex = new RegExp('chrome://extensions');
+                        if (matchRegexToJsRegex(domainName).test(tab.url) && (!chromeRegex.test(tab.url))) {
+                            chrome.tabs.reload(tab.id);
+                        }
+                    });
+                    saveTemplates(domainJSON, callback, domainID);
+                });
+            }
+        });
+    });
+}
+
+function removeDomain (domainName, removeAll, callback) {
+    var domainID;
+    domainName = domainName.trim();
+    chrome.storage.sync.get(StorageID, function (data) {
+        if (data[StorageID]) {
+            var domainJSON = data[StorageID];
+            var domains = data[StorageID].options.domains;
+            $.each(domains, function (index, domain) {
```

Comment:
We should just be using the domainID. It would simplify this greatly.

Reply:
do you mean using the domainJSON? Or do you mean obtaining the domainID earlier, and then passing it to removeDomain instead of domainName?

If the latter, we would still have to loop through the domains to find the ID, it would just happen somwhere else.

Reply:
When we build the domain list, if we embed the id of the domain into either the trash icon tag as a data attr or something, then when the user clicks the "trash" icon we can simply look up the id and pass that through.